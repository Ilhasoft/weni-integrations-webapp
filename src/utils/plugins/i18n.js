import Vue from 'vue';
import VueI18n from 'vue-i18n';

Vue.use(VueI18n);

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';
import es_es from '../../locales/es_es.json';

const languages = {
  'en-us': en,
  'pt-br': pt_br,
  es: es_es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en-us',
  fallbackLocale: 'en-us',
  messages,
});

export default i18n;
