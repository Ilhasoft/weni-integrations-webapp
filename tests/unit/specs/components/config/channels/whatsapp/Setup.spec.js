import { unnnicCallAlert as mockUnnnicCallAlert } from '@weni/unnnic-system';

jest.mock('@weni/unnnic-system', () => ({
  ...jest.requireActual('@weni/unnnic-system'),
  unnnicCallAlert: jest.fn(),
}));

import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import wppCloudSetup from '@/components/config/channels/whatsapp/Setup.vue';
import i18n from '@/utils/plugins/i18n';
import { singleApp } from '../../../../../../__mocks__/appMock';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('whatsapp/Setup.vue', () => {
  let wrapper;
  let state;
  let wppCloudState;
  let wppCloudActions;
  let store;

  beforeEach(() => {
    const customData = {
      input_token: '123',
    };

    state = {
      auth: {
        project: '123',
      },
    };

    wppCloudState = {
      wabaId: null,
      businessId: null,
      whatsAppPhoneNumbers: [{ display_phone_number: '123', verified_name: 'number', id: 1 }],
      selectedPhoneNumber: null,

      loadingDebugToken: false,
      errorDebugToken: false,

      loadingPhoneNumbers: false,
      errorPhoneNumbers: false,

      loadingWhatsAppCloudConfigure: false,
      errorCloudConfigure: false,
    };

    wppCloudActions = {
      setSelectedPhoneNumber: jest.fn(),
      configurePhoneNumber: jest.fn(),
      getDebugToken: jest.fn(() =>
        Promise.resolve({
          data: {
            waba_id: 'waba_id_123',
            business_id: 'business_id_123',
          },
        }),
      ),
      getWhatsAppPhoneNumbers: jest.fn(() =>
        Promise.resolve({
          data: [{ id: 1 }, { id: 2 }],
        }),
      ),
    };

    store = new Vuex.Store({
      modules: {
        WhatsAppCloud: {
          namespaced: true,
          actions: wppCloudActions,
          state: wppCloudState,
        },
      },
      state,
    });

    wrapper = shallowMount(wppCloudSetup, {
      localVue,
      i18n,
      store,
      propsData: {
        app: singleApp,
        customData,
      },
      mocks: {
        $t: () => 'some specific text',
        $router: {
          replace: jest.fn(),
        },
        $route: {
          path: '/apps/1/details',
        },
      },
      stubs: {
        UnnnicModal: true,
        UnnnicButton: true,
        UnnnicSelect: true,
      },
    });
  });

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should be rendered properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('closePopUp()', () => {
    it('should call parent closePopUp()', () => {
      expect(wrapper.emitted('closePopUp')).toBeFalsy();
      wrapper.vm.closePopUp();
      expect(wrapper.emitted('closePopUp')).toBeTruthy();
    });
  });

  describe('changeLoginState', () => {
    it('should set onLogin to desired state', () => {
      expect(wrapper.vm.onLogin).toEqual(false);
      wrapper.vm.changeLoginState(true);
      expect(wrapper.vm.onLogin).toEqual(true);
    });
  });

  describe('createChannel()', () => {
    it('should change url into /apps/my', async () => {
      const spy = spyOn(wrapper.vm.$router, 'replace');
      await wrapper.vm.createChannel();

      expect(spy).toBeCalledTimes(1);
      expect(spy).toHaveBeenCalledWith(`/apps/my`);
    });

    it('should call configurePhoneNumber', async () => {
      expect(wppCloudActions.configurePhoneNumber).not.toHaveBeenCalled();
      await wrapper.vm.createChannel();
      expect(wppCloudActions.configurePhoneNumber).toHaveBeenCalledTimes(1);
    });

    it('should call callErrorModal on error', async () => {
      store.state.WhatsAppCloud.errorCloudConfigure = true;
      const spy = spyOn(wrapper.vm, 'callErrorModal');
      expect(spy).not.toHaveBeenCalled();
      await wrapper.vm.createChannel();
      expect(spy).toHaveBeenCalledTimes(1);
    });

    it('should not call configurePhoneNumber if loadingWhatsAppCloudConfigure is defined', async () => {
      store.state.WhatsAppCloud.loadingWhatsAppCloudConfigure = true;
      expect(wppCloudActions.configurePhoneNumber).not.toHaveBeenCalled();
      await wrapper.vm.createChannel();
      expect(wppCloudActions.configurePhoneNumber).not.toHaveBeenCalled();
    });
  });

  describe('callErrorModal', () => {
    it('should call unnnicCallAlert', () => {
      expect(mockUnnnicCallAlert).not.toHaveBeenCalled();
      wrapper.vm.callErrorModal({ text: 'error text' });
      expect(mockUnnnicCallAlert).toHaveBeenCalledTimes(1);
      expect(mockUnnnicCallAlert).toHaveBeenCalledWith(
        expect.objectContaining({
          props: {
            text: 'error text',
            title: expect.any(String),
            icon: expect.any(String),
            scheme: expect.any(String),
            position: expect.any(String),
            closeText: expect.any(String),
          },
          seconds: expect.any(Number),
        }),
      );
    });
  });
});
