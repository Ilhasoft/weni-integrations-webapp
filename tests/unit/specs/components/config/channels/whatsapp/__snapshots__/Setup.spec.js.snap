// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`whatsapp/Setup.vue should be rendered properly 1`] = `
<div customdata="[object Object]">
  <unnnicmodal-stub text="some specific text" modalicon="phone-3" description="some specific text" showmodal="true" scheme="feedback-green" class="whatsapp-setup">
    <div>
      <div class="whatsapp-setup__buttons">
        <unnnicbutton-stub size="large" text="some specific text" type="terciary" class="whatsapp-setup__buttons__cancel"></unnnicbutton-stub>
        <loadingbutton-stub type="secondary" size="large" loadingposition="left" text="some specific text" class="whatsapp-setup__buttons__start"></loadingbutton-stub>
      </div>
    </div>
  </unnnicmodal-stub>
</div>
`;
